eth_tester/utils/throws_contract.py

```diff
@@ -29,46 +40,78 @@
+    "01000a031916815260200191505b509250505060405180910390f35b60608115610164576040517f0"
+    "8c379a000000000000000000000000000000000000000000000000000000000815260040180806020"
+    "01828103825260108152602001807f726962626572742c20726962626572740000000000000000000"
+    "000000000000081525060200191505060405180910390fd5b6040805190810160405280600a815260"
+    "20017f4e6f20726962626572740000000000000000000000000000000000000000000081525090509"
+    "190505600a165627a7a723058206fc7f85b6a52373bc74978757e75e3a100b44024e61485612d0f3c"
+    "d27e32cdd60029"
 )
 
 
-THROWS_ABI = {
-    'willThrow': {
-        'constant': False,
-        'inputs': [],
-        'name': 'willThrow',
-        'outputs': [],
-        'payable': False,
-        'stateMutability': 'nonpayable',
-        'type': 'function',
-    },
-    'value': {
-        'constant': True,
-        'inputs': [],
-        'name': 'value',
-        'outputs': [{'name': '', 'type': 'uint256'}],
-        'payable': False,
-        'stateMutability': 'view',
-        'type': 'function',
-    },
-    'setValue': {
-        'constant': False,
-        'inputs': [{'name': '_value', 'type': 'uint256'}],
-        'name': 'setValue',
-        'outputs': [],
-        'payable': False,
-        'stateMutability': 'nonpayable',
-        'type': 'function',
+THROWS_ABI = (
+    {
+        'willThrow': {
+            'constant': False,
+            'inputs': [],
+            'name': 'willThrow',
+            'outputs': [],
+            'payable': False,
+            'stateMutability': 'nonpayable',
+            'type': 'function',
+        },
+        'value': {
+            'constant': True,
+            'inputs': [],
+            'name': 'value',
+            'outputs': [{'name': '', 'type': 'uint256'}],
+            'payable': False,
+            'stateMutability': 'view',
+            'type': 'function',
+        },
+        'setValue': {
+            'constant': False,
+            'inputs': [{'name': '_value', 'type': 'uint256'}],
+            'name': 'setValue',
+            'outputs': [],
+            'payable': False,
+            'stateMutability': 'nonpayable',
+            'type': 'function',
+        }
     },
-}
+    {
+        'do_revert': {
+            "constant": True,
+            "inputs": [{
+                "name": "_revert",
+                "type": "bool"
+            }],
+            "name": "do_revert",
+            "outputs": [{
+                "name": "message",
+                "type": "string"
+            }],
+            "payable": False,
+            "stateMutability": "pure",
+            "type": "function"
+        }
+    }
+)
 
 
-def _deploy_throws(eth_tester):
+def _deploy_throws(eth_tester, contract_idx=0):
     deploy_hash = eth_tester.send_transaction({
         "from": eth_tester.get_accounts()[0],
         "gas": 500000,
-        "data": THROWS_BYTECODE,
+        "data": THROWS_BYTECODE[contract_idx],
```

Comment:
Can we drop the `contract_idx` in favor of distinct contracts?

Reply:
Sure, I was initially going to create own `revert_contract.py` but then realised the _throw function would be exact replicas - which lead me to making multiple contracts.

What other type of identifier could I use on the function call level, assuming I keep the _*throws functions (dict instead of tuple with contract_name arg)?